0.18.0: unable to run imapwidget related tests

# Issue description

When running the tests for 0.18.0 I came across a set of tests that always fail for me and I can't seem to get them to work.
I have reviewed https://github.com/qtile/qtile/blob/4f398189420b5020a4403a3556d4e4a98db38982/.github/workflows/ci.yml, https://github.com/qtile/qtile/blob/4f398189420b5020a4403a3556d4e4a98db38982/tox.ini and https://github.com/qtile/qtile/blob/4f398189420b5020a4403a3556d4e4a98db38982/setup.cfg to figure out the probable dependencies.
A lot has changed since 0.17.0 and the CI integration doesn't clearly distinguish between what is required for qtile to run/to be tested and what is required to build the additional dependencies (this is *very* time consuming).

```
FAILED test/widgets/test_imapwidget.py::test_imapwidget - ConnectionRefusedEr...
FAILED test/widgets/test_imapwidget.py::test_imapwidget_keyring_error - Conne...
FAILED test/widgets/test_imapwidget.py::test_imapwidget_password_none - Conne...
FAILED test/widgets/test_widget_init_configure.py::test_widget_init_config[ImapWidget-kwargs28]
===== 4 failed, 582 passed, 7 skipped, 27 deselected in 124.12s (0:02:04) ======
```

I have tried to use libgnome-keyring, kwallet + python-dbus, python-keyrings-alt in combination with python-keyring (which seems to be a dependency of the widget), all to no avail.
As I neither use e.g. gnome-keyring or kwallet nor could find information about how the expected setup in CI is handling this (e.g. running a dbus session?), I have no way of figuring out how to prepare the environment unfortunately.

# Qtile version

0.18.0

# Stack traces

Full build logs:
- [qtile-0.18.0-build.log](https://github.com/qtile/qtile/files/6835127/qtile-0.18.0-build.log) (python-keyrings-alt, python-fs, python-cryptodome)
- [qtile-0.18.0-build-kwallet.log](https://github.com/qtile/qtile/files/6835141/qtile-0.18.0-build-kwallet.log) (python-dbus, kwallet)
- [qtile-0.18.0-build-libgnome-keyring.log](https://github.com/qtile/qtile/files/6835147/qtile-0.18.0-build-libgnome-keyring.log) (libgnome-keyring)

In the latter two cases python-keyring seems unable to even find the backends (although they should be there).

# Configuration

n/a

--- a/test/widgets/test_imapwidget.py
+++ b/test/widgets/test_imapwidget.py
@@ -21,6 +21,7 @@
 # Widget specific tests
 
 import sys
+from importlib import reload
 from types import ModuleType
 
 import pytest
@@ -68,12 +69,19 @@ class FakeKeyring(ModuleType):
             return None
 
 
-def test_imapwidget(fake_qtile, monkeypatch, fake_window):
+@pytest.fixture()
+def patched_imap(monkeypatch):
+    monkeypatch.delitem(sys.modules, "imaplib", raising=False)
+    monkeypatch.delitem(sys.modules, "keyring", raising=False)
     monkeypatch.setitem(sys.modules, "imaplib", FakeIMAP("imaplib"))
     monkeypatch.setitem(sys.modules, "keyring", FakeKeyring("keyring"))
     from libqtile.widget import imapwidget
+    reload(imapwidget)
+    yield imapwidget
 
-    imap = imapwidget.ImapWidget(user="qtile")
+
+def test_imapwidget(fake_qtile, monkeypatch, fake_window, patched_imap):
+    imap = patched_imap.ImapWidget(user="qtile")
     fakebar = Bar([imap], 24)
     fakebar.window = fake_window
     fakebar.width = 10
@@ -84,13 +92,9 @@ def test_imapwidget(fake_qtile, monkeypatch, fake_window):
     assert text == "INBOX: 2"
 
 
-def test_imapwidget_keyring_error(fake_qtile, monkeypatch, fake_window):
-    monkeypatch.setitem(sys.modules, "imaplib", FakeIMAP("imaplib"))
-    monkeypatch.setitem(sys.modules, "keyring", FakeKeyring("keyring"))
-    from libqtile.widget import imapwidget
-    imapwidget.keyring.valid = False
-
-    imap = imapwidget.ImapWidget(user="qtile")
+def test_imapwidget_keyring_error(fake_qtile, monkeypatch, fake_window, patched_imap):
+    patched_imap.keyring.valid = False
+    imap = patched_imap.ImapWidget(user="qtile")
     fakebar = Bar([imap], 24)
     fakebar.window = fake_window
     fakebar.width = 10
@@ -106,14 +110,11 @@ def test_imapwidget_keyring_error(fake_qtile, monkeypatch, fake_window):
 # The widget will then fail when it looks up this attribute and then
 # fail again when it tries to return self.text.
 # TO DO: Fix widget's handling of this scenario.
-def test_imapwidget_password_none(fake_qtile, monkeypatch, fake_window):
-    monkeypatch.setitem(sys.modules, "imaplib", FakeIMAP("imaplib"))
-    monkeypatch.setitem(sys.modules, "keyring", FakeKeyring("keyring"))
-    from libqtile.widget import imapwidget
-    imapwidget.keyring.valid = False
-    imapwidget.keyring.error = False
+def test_imapwidget_password_none(fake_qtile, monkeypatch, fake_window, patched_imap):
+    patched_imap.keyring.valid = False
+    patched_imap.keyring.error = False
 
-    imap = imapwidget.ImapWidget(user="qtile")
+    imap = patched_imap.ImapWidget(user="qtile")
     fakebar = Bar([imap], 24)
     fakebar.window = fake_window
     fakebar.width = 10

